rules/gcp_k8s_rules/gcp_k8s_exec_into_pod.py

```diff
@@ -1,37 +1,42 @@
 from gcp_base_helpers import get_k8s_info
 from gcp_environment import PRODUCTION_PROJECT_IDS, rule_exceptions
-from panther_base_helpers import deep_get
+from panther_base_helpers import deep_walk
 
 
 def rule(event):
     # Defaults to False (no alert) unless method is exec and principal not allowed
-    if not deep_get(
+    if not deep_walk(
         event, "protoPayload", "methodName"
-    ) == "io.k8s.core.v1.pods.exec.create" or not deep_get(event, "protoPayload", "resourceName"):
+    ) == "io.k8s.core.v1.pods.exec.create" or not deep_walk(event, "protoPayload", "resourceName"):
         return False
 
     k8s_info = get_k8s_info(event)
-    principal = k8s_info["principal"]
-    namespace = k8s_info["namespace"]
-    project_id = k8s_info["project_id"]
+    principal = deep_walk(k8s_info, "principal", default="")
+    namespace = deep_walk(k8s_info, "namespace", default="")
+    project_id = deep_walk(k8s_info, "project_id", default="")
     # rule_exceptions that are allowed temporarily are defined in gcp_environment.py
     # Some execs have principal which is long numerical UUID, appears to be k8s internals
-    for allowed_principal in rule_exceptions["gcp_k8s_exec_into_pod"]["allowed_principals"]:
+    for allowed_principal in deep_walk(
+        rule_exceptions, "gcp_k8s_exec_into_pod", "allowed_principals", default=""
+    ):
         if (
-            principal in allowed_principal["principals"]
+            principal in deep_walk(allowed_principal, "principals", default="")
             and (
-                not allowed_principal["namespaces"] or namespace in allowed_principal["namespaces"]
+                not deep_walk(allowed_principal, "namespaces", default="")
+                or namespace in deep_walk(allowed_principal, "namespaces", default="")
+            )
+            and (
+                not deep_walk(allowed_principal, "projects", default="")
+                or project_id in deep_walk(allowed_principal, "projects", default="")
```

Comment:
I think this can also collapse if the default value is the empty list

```suggestion
                project_id in deep_walk(allowed_principal, "projects", default=[])
```